tt_metal/impl/kernels/runtime_args_data.hpp

```diff
@@ -16,19 +21,60 @@ struct RuntimeArgsData {
     std::uint32_t * rt_args_data;
     std::size_t rt_args_count;
 
-    std::uint32_t & operator[](std::size_t index);
+    inline bool in_bounds(std::size_t index) const noexcept {
+        if(index >= rt_args_count) [[unlikely]] {
+            std::cerr << "TT_FATAL: Index " << index << " is larger than runtime args size "
+                      << rt_args_count << " at " << __FILE__ << ":" << __LINE__ << std::endl;
+            return false;
+        }
+        return true;
+    }
 
-    const std::uint32_t& operator[](std::size_t index) const;
+    inline bool in_bounds_or_throw(std::size_t index) const {
+        if(!in_bounds(index)) [[unlikely]] {
+            throw std::out_of_range(
+                "Index " + std::to_string(index) + " is larger than runtime args size " + std::to_string(rt_args_count)
+            );
+        }
+        return true;
+    }
 
-    std::uint32_t & at(std::size_t index);
+    inline std::uint32_t & operator[](std::size_t index) noexcept {
+        assert(in_bounds(index));
+        return this->rt_args_data[index];
+    }
 
-    const std::uint32_t& at(std::size_t index) const;
+    inline const std::uint32_t& operator[](std::size_t index) const noexcept {
+        assert(in_bounds(index));
+        return this->rt_args_data[index];
+    }
 
-    std::uint32_t * data() noexcept;
+    inline std::uint32_t & at(std::size_t index) {
+        if (in_bounds_or_throw(index)) [[likely]] {
```

Comment:
btw it is bette to do check if (something is bad) throw, compiler automatically recognizes the pattern and you don't need to add likely/unlikely.

Reply:
ok, I'll adjust

Reply:
Thanks for the suggestion, I like it better now.